src/store/modules/layers.store.js

```diff
@@ -78,6 +78,12 @@ const state = {
      * @type AbstractLayer[]
      */
     systemLayers: [],
+    /**
+     * Name of the active KML layer for drawing.
+     *
+     * @type {String | null}
+     */
+    activeKmlLayerName: null,
```

Comment:
Not sure you need to store that, it's modified only in the context of the DrawingModule (or more precisely in DrawingToolbox) and I don't think it needs to be "accessed" or changed in any other way than reading the KML metadata later on.

IMHO you could remove this from the store

Reply:
The problem is when there is no active kml layer yet but you already want to save the name the name has to be stored somewhere so it won't work to remove this variable from the store

Reply:
Cant' you keep track of it in the component only?
If it's not possible to have it any other place than the store, I'd move it to `drawing.store.js` to clarify what is the purpose of this (and maybe name it, `drawingName` or something like that)

Reply:
I think it depends on the desired behaviour
for example what should be the behaviour when you change the name, click on validate, then return to the main menu without having drawn anything. Should then be the saved name be there when you return to the drawing interface or not?